src/geo/map/cartodb-layer.js

```diff
@@ -147,6 +147,16 @@ var CartoDBLayer = LayerModelBase.extend({
   remove: function () {
     this.getSource().unmarkAsSourceOf(this);
     LayerModelBase.prototype.remove.apply(this, arguments);
+  },
+
+  $getTableName: function () {
```

Comment:
I would add some tests for `$getTableName` and `$getApiKey`. 

In addition, but this is a **personal** opinion, I would return `null` if there are not `this.attributes.source.attributes.options`. If the result of this function is assigned to a value, this value would be `undefined`, and I consider that is more correct to set it to `null` instead of `undefined`.